rapids-cmake/test/detail/generate_resource_spec/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+#=============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+cmake_minimum_required(VERSION 3.23.1)
+project(generate_resource_spec ${lang})
+
+set(CMAKE_CUDA_ARCHITECTURES all)
+
+add_executable(generate_ctest_json generate_resource_spec.cpp)
+target_link_libraries(generate_ctest_json PRIVATE rt ${CMAKE_DL_LIBS})
+if(cuda_toolkit)
+  find_package(CUDAToolkit REQUIRED QUIET)
+  target_link_libraries(generate_ctest_json PRIVATE CUDA::cudart)
```

Comment:
Link to `CUDA::cudart_static` so we mantain not having a runtime dependency on the CUDA toolkit.

You will also need to set the `CUDA_RUNTIME_LIBRARY` target property to `STATIC` as well